libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt

```diff
@@ -50,11 +52,17 @@ object Route {
         query.execute(waypoints_)
     }
 
-    fun LiveReplanRoute(plan: String,
-                        speed: Int,
+    fun LiveReplanRoute(speed: Int,
                         context: Context,
                         waypoints: Waypoints) {
-        val query = LiveRideReplanRoutingTask(plan, speed, context)
+        var newPlan: String
+        // Check current plan is a linear route plan.
+        if (currentJourneyPlan in RoutePlans.allPlans())
+            newPlan = currentJourneyPlan
+        else
+            newPlan = RoutePlans.PLAN_QUIETEST
```

Comment:
As discussed I think this should be Balanced.

Reply:
I'm happy with Balanced, but it depends on whether Circular routes are consistently planned as quietest or balanced at the back end, in order to stick as closely as possible to the original route.
Hopefully @mvl22 can tell us that.

Reply:
I made the change, then saw Martin's comment to say Circular routes are based on Quietest, so I've taken the change out again!
I've also changed Boolean to a primitive in RouteData.